core/src/main/scala/kafka/server/DelayedFetch.scala

```diff
@@ -93,14 +93,14 @@ class DelayedFetch(
               if (endOffset.onOlderSegment(fetchOffset)) {
                 // Case F, this can happen when the new fetch operation is on a truncated leader
                 debug(s"Satisfying fetch $this since it is fetching later segments of partition $topicIdPartition.")
-                return forceComplete()
+                forceComplete()
```

Comment:
Does this have the same semantics if forceComplete returns false?

Reply:
no, good point. Let me take a look and see if I can address this.